src/parser.rs

```diff
@@ -283,6 +307,17 @@ impl Pattern for char {
     }
 }
 
+impl Pattern for String {
```

Comment:
Isn't `Pattern for &'_ str` enough? Why do we need this impl?

Reply:
Yeah It's because of the useless shenanigans I pulled with the after_prefix. Just removed it.